### ISSUE TITLE
Still HTTPS issue

### ISSUE THREAD

**Issue by MESWEB** [piwik/piwik#11478]
PIWIK messing security of the website
![zrzut ekranu 2017-03-12 13 08 33](https://cloud.githubusercontent.com/assets/2931024/23832047/4a8dbef4-0725-11e7-8865-1efc89fa354f.png)

**Comment by Findus23** [comment]
@MESWEB 
There is something wrong with your webserver configuration:

The tracking request `https://piwik.auctio.uk/piwik.php?action_name=auctio.uk&idsite=1&rec=1&r=695582&h=14&m=15&s=20&url=https%3A%2F%2Fauctio.uk%2F&_id=ca718d81d71254bf&_idts=1489324450&_idvc=1&_idn=0&_refts=0&_viewts=1489324450&send_image=1&pdf=1&qt=0&realp=0&wma=0&dir=0&fla=0&java=0&gears=0&ag=0&cookie=1&res=1920x1080&gt_ms=197&pv_id=fxXl6Q`
gets redirected to `http://piwik.auctio.uk/`
which redirects to `https://piwik.auctio.uk/`

**Comment by MESWEB** [comment]
So what I should to ask my hosting support?

**Comment by Findus23** [comment]
I can't test anymore as `http://auctio.uk/` doesn't load anymore. (seems like my IP got blocked), but there seems to be a wrong redirect (maybe in a .htaccess-file?)

**Comment by Findus23** [comment]
Now all my IPs are blocked but it seems like the tracking code is correct. Is it possible that there is some regex that matches some text in the GET parameters? (maybe `action_name=auctio.uk`)

But it still doesn't make sense that it redirects to the http site.

**Comment by MESWEB** [comment]
No. I was contact with host support and we found issue with PIWIK script. Piwik sending request by http: like display images in dashboard with http://

**Comment by Findus23** [comment]
Could you be more specific what exactly the http request is?

I am still blocked, but I have tested it via my phone and this are the requests:
![screenshot_20170312_214108](https://cloud.githubusercontent.com/assets/6266037/23835731/a9e2bc5a-076c-11e7-82ac-76217a441472.png)
At first the piwik js-file is loaded (via https): `https://piwik.auctio.uk/piwik.js`.
Then a visit is tracked via the tracking-pixel `https://piwik.auctio.uk/piwik.php?idsite=1&rec=1`
I guess you'll want to remove the `<img>` or wrap it with a `<noscript>` as it is only needed when javascript is disabled.
```html
<!-- End Piwik Code --><!-- Piwik Image Tracker-->
<img src="//piwik.auctio.uk/piwik.php?idsite=1&rec=1" style="border:0" alt="" />
<!-- End Piwik --><!-- jQuery (necessary for Bootstrap's JavaScript plugins) -->
```

Next the piwik tracking script sends the data back (**via HTTPS**)
`https://piwik.auctio.uk/piwik.php?action_name=auctio.uk&idsite=1&rec=1&r=354129&h=21&m=38&s=15&url=https%3A%2F%2Fauctio.uk%2F&_id=e3374b7377e825bf&_idts=1489351095&_idvc=1&_idn=0&_refts=0&_viewts=1489351095&send_image=1&pdf=1&qt=0&realp=0&wma=0&dir=0&fla=0&java=0&gears=0&ag=0&cookie=1&res=1920x1080&gt_ms=234&pv_id=KiV8jM`

But instead of saving the data, the server response with a 302-redirect to **http**://piwik.auctio.uk/. (Because of that I assume the visit won't show up in Piwik (or just because of the image tracker)) 

```http
HTTP/1.1 302 Found
Date: Sun, 12 Mar 2017 20:38:13 GMT
Server: Apache
Location: http://piwik.auctio.uk/
Content-Length: 207
Keep-Alive: timeout=5, max=99
Connection: Keep-Alive
Content-Type: text/html; charset=iso-8859-1
```

The browser follows the redirect to **http**://piwik.auctio.uk/ but again gets another (this time permanent redirect to https://auctio.uk/:

```http
HTTP/1.1 301 Moved Permanently
Date: Sun, 12 Mar 2017 20:38:13 GMT
Server: Apache
Location: https://auctio.uk/
Content-Length: 226
Keep-Alive: timeout=5, max=100
Connection: Keep-Alive
Content-Type: text/html; charset=iso-8859-1
```

To sum up: The piwik tracking tries to send the data via https, but gets redirected to the homepage via http which causes chrome to complain.

**Comment by MESWEB** [comment]
Yes but I have problem with images on dashboard
![zrzut ekranu 2017-03-12 22 09 01](https://cloud.githubusercontent.com/assets/2931024/23836462/907b79ec-0770-11e7-92be-cfdf6b2374f6.png)
So I think piwik.js have bug with https.

**Comment by s1awa** [comment]
@MESWEB I use piwik with https and have no issues. It is definitely your setup that does wrong redirects which break piwik.

check your url rewrite rules , etc.

When I open http://auctio.uk I get redirected to https://https//auctio.uk//

**Comment by MESWEB** [comment]
@s1awa Ok. Sorry for that. Now is working I was change .htaccess.

**Comment by MESWEB** [comment]
Problem fixed by:
**disabled mod_security**
